src/nvim/api/private/helpers.c

```diff
@@ -808,6 +811,17 @@ static void init_type_metadata(Dictionary *metadata)
 
   PUT(*metadata, "types", DICTIONARY_OBJ(types));
 }
+static void init_api_level_metadata(Dictionary *metadata)
+{
+  Dictionary version = ARRAY_DICT_INIT;
+  PUT(version, "current", INTEGER_OBJ(NVIM_API_CURRENT));
+  PUT(version, "compatibility", INTEGER_OBJ(NVIM_API_COMPATIBILITY));
+#ifdef NVIM_VERSION_PRERELEASE
+  PUT(version, "prerelease", BOOLEAN_OBJ(true));
```

Comment:
Thanks for your patience in explaining your perspective.

What I had been considering is the flow of our development and someone testing out that development.  In the proposed scenario, that means that we bump the API level when we make an API change during our development cycle.

What I hadn't been thinking of was the perspective of a plugin author trying to protect someone from running a plugin using "stable API level X" with an nvim that has "in-development API level X".

With that lens, I see two options.  Either wait to bump the API level until the release happens (which has a higher cognitive burden on our end) or introduce a "pre-release" indicator as @bfredl has suggested.  That would still need to be toggled before release, but that's easier to catch/automate.